framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -285,104 +283,101 @@ protected boolean areEqualInHierarchy(
       return true;
     }
-
-    if (outside.getKind() != TypeKind.WILDCARD
-        && !TypesUtils.isCaptured(outside.getUnderlyingType())) {
-      if (canBeCovariant) {
-        return isSubtype(inside, outside, currentTop);
+    if (outside.getKind() == TypeKind.WILDCARD
+        // TODO: the below should be removed after #979 is fixed.
+        || (TypesUtils.isCapturedTypeVariable(outside.getUnderlyingType())
+            && !TypesUtils.isCapturedTypeVariable(inside.getUnderlyingType()))) {
+
+      Boolean previousResult = areEqualVisitHistory.get(inside, outside, currentTop);
+      if (previousResult != null) {
+        return previousResult;
+      }
+      areEqualVisitHistory.put(inside, outside, currentTop, true);
+      boolean result;
+      if (outside.getKind() == TypeKind.WILDCARD) {
+        AnnotatedWildcardType outsideWildcard = (AnnotatedWildcardType) outside;
+        result =
+            isContainedByBoundType(
+                inside,
+                outsideWildcard.getSuperBound(),
+                outsideWildcard.getExtendsBound(),
+                canBeCovariant);
+      } else {
+        AnnotatedTypeVariable outsideTypeVar = (AnnotatedTypeVariable) outside;
+        result =
+            isContainedByBoundType(
+                inside,
+                outsideTypeVar.getLowerBound(),
+                outsideTypeVar.getUpperBound(),
+                canBeCovariant);
       }
-      return areEqualInHierarchy(inside, outside);
-    }
 
-    AnnotatedTypeMirror outsideUpperBound;
-    AnnotatedTypeMirror outsideLowerBound;
-    if (outside.getKind() == TypeKind.WILDCARD) {
-      outsideUpperBound = ((AnnotatedWildcardType) outside).getExtendsBound();
-      outsideLowerBound = ((AnnotatedWildcardType) outside).getSuperBound();
-    } else if (TypesUtils.isCaptured(outside.getUnderlyingType())) {
-      outsideUpperBound = ((AnnotatedTypeVariable) outside).getUpperBound();
-      outsideLowerBound = ((AnnotatedTypeVariable) outside).getLowerBound();
-    } else {
-      throw new BugInCF("Expected a wildcard or captured type variable, but found " + outside);
+      areEqualVisitHistory.put(inside, outside, currentTop, result);
+      return result;
     }
-    Boolean previousResult = areEqualVisitHistory.get(inside, outside, currentTop);
-    if (previousResult != null) {
-      return previousResult;
+    if (canBeCovariant) {
+      return isSubtype(inside, outside, currentTop);
     }
-
-    areEqualVisitHistory.put(inside, outside, currentTop, true);
-    boolean result =
-        isContainedWildcard(inside, outside, outsideUpperBound, outsideLowerBound, canBeCovariant);
-    areEqualVisitHistory.put(inside, outside, currentTop, result);
-    return result;
+    return areEqualInHierarchy(inside, outside);
   }
 
-  private boolean isContainedWildcard(
+  /**
+   * Returns true if {@code outside} contains {@code inside}, that is, if the set of types denoted
```

Comment:
There is no variable `outside`.  I think this comment should start with:

Let `outside` be `? super outsideLower extends outsideUpper`.